src/UI/researchScreen/ResearchScreen.ts

```diff
@@ -18,8 +19,15 @@ export default class ResearchScreen implements Page {
     refresh(): void {
         let researchHeader = UI.makeHeader("Available Research Projects");
 
-
         const possibleTechs: Technology[] = this.run.getResearchOptions();
+        const researchResources: Resource[] = Resource.values().filter(resource => possibleTechs.filter(tech => tech.researchCost.resource == resource).length != 0);
```

Comment:
There's a more readable way to do this. Instead of doing a filter and then checking the length, you should use replace `possibleTechs.filter` with `possibleTechs.some`, which returns a boolean based on whether at least one element in the array passes the given test. See [the MDN doc on Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).